src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java

```diff
@@ -171,20 +173,33 @@ public boolean exists(String datasetName) throws SQLEngineException {
 
   @Override
   public boolean canJoin(SQLJoinDefinition sqlJoinDefinition) {
-    // TODO: remove when expression joins are supported.
-    if (sqlJoinDefinition.getJoinDefinition().getCondition().getOp() != JoinCondition.Op.KEY_EQUALITY) {
-      return false;
-    }
 
-    //Ensure none of the schemas contains unsupported types.
-    for (JoinStage inputStage : sqlJoinDefinition.getJoinDefinition().getStages()) {
+    List<String> issues = new LinkedList<>();
```

Comment:
How is this issues list used? I see you add data to it, but not retrieve. Also please use ArrayList. LinkedList makes sense only in very edge cases. In most cases ArrayList is faster and use less memory.

Reply:
Oh, good catch, I must have deleted the block where I print out a warning. Let me add it again.